src/crdt.rs

```diff
@@ -1513,6 +1516,30 @@ mod tests {
         assert!(one && two);
     }
 
+    #[test]
+    fn gossip_request_bad_addr() {
+        let me = NodeInfo::new(
```

Comment:
Can you use your `new_localhost` constructor here?

Reply:
thats on TestNode, and this thing takes already created addresses with ports.  TestNode generates the local host addresses by binding and then asking what port the OS allocated.